test-data-generator/src/main/scala/EventGenerator.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import za.co.absa.spline.producer.model.v1_2.{ExecutionEvent, ExecutionPlan}
+
+object EventGenerator {
+
+  def generate(executionPlan: ExecutionPlan): ExecutionEvent = {
+    ExecutionEvent(
+      planId = executionPlan.id,
+      timestamp = System.currentTimeMillis(),
+      durationNs = None,
+      error = None,
+      extra = Map()
```

Comment:
nit: `Map.empty` is more conventional FP-style notation, while `Map()` looks more natural for imperative/mutable style as it resembles creation of a new instance of `Map'.